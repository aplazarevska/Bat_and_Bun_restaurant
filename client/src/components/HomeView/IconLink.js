import React from 'react';
import { Paper, Button, Avatar, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

const Item = styled(Paper)(({ theme }) => ({
  ...theme.typography.body2,
  textAlign: 'center',
  height: 150,
  lineHeight: '150px',
}));

const IconLink = ({ linkName, linkUrl, linkIcon }) => {
  return (
    <Item
      data-testid="link-icon"
      elevation={0}
      className="link-icon"
      component={Button}
      disableRipple
      size="small"
      target="_blank"
      rel="noreferrer noopener"
      href={linkUrl}
    >
      <Avatar alt={linkName} src={linkIcon} variant="square" sx={{ width: 64, height: 64 }} />
      <Typography data-testid="link-name" variant="h6">{linkName}</Typography>
    </Item>
  )
}

export default IconLink;
