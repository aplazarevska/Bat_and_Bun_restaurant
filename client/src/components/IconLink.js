import { Link } from 'react-router-dom';
import { Paper, Avatar, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

const Item = styled(Paper)(({ theme }) => ({
  ...theme.typography.body2,
  textAlign: 'center',
  height: 150,
  lineHeight: '150px',
}));

const IconLink = ({ linkName, linkUrl, linkIcon }) => {
  return (
    <Item elevation={1} className="link-icon" component={Link} to={linkUrl}>
      <Avatar alt={linkName} src={linkIcon} variant="square" sx={{ width: 64, height: 64 }} />
      <Typography variant="h6">{linkName}</Typography>
    </Item>
  )
}

export default IconLink;
