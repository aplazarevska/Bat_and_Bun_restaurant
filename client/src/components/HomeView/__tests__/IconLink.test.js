import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import IconLink from '../IconLink';
import { afterEach, expect } from '@jest/globals';

const linkName = 'test link name';
const linkUrl = 'https://google.com';
const linkIcon = './img/uber-eats-icon.png';

afterEach(() => {
  cleanup();
});

test("Should render icon link component that has a href", () => {
  render(<BrowserRouter><IconLink linkName={linkName} linkUrl={linkUrl} linkIcon={linkIcon} /></BrowserRouter>);
  const iconLinkElement = screen.getByTestId('link-icon');
  expect(iconLinkElement).toBeInTheDocument();
  expect(iconLinkElement).toHaveAttribute('href', 'https://google.com');
});

test("Should render icon link component that has an icon image", () => {
  render(<BrowserRouter><IconLink linkName={linkName} linkUrl={linkUrl} linkIcon={linkIcon} /></BrowserRouter>);
  const iconImageElement = screen.getByRole('img');
  expect(iconImageElement).toHaveAttribute('src', './img/uber-eats-icon.png');
});

test("Should render icon link component that has a link name", () => {
  render(<BrowserRouter><IconLink linkName={linkName} linkUrl={linkUrl} linkIcon={linkIcon} /></BrowserRouter>);
  const linkNameElement = screen.getByTestId('link-name');
  expect(linkNameElement).toHaveTextContent('test link name');
});

test("Matches snapshot", () => {
  const tree = renderer.create(<BrowserRouter><IconLink linkName={linkName} linkUrl={linkUrl} linkIcon={linkIcon} /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
