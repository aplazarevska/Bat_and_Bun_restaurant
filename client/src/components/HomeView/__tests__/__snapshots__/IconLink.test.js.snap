// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<a
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 link-icon css-xmlano-MuiButtonBase-root-MuiButton-root-MuiPaper-root"
  data-testid="link-icon"
  href="https://google.com"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  rel="noreferrer noopener"
  tabIndex={0}
  target="_blank"
>
  <div
    className="MuiAvatar-root MuiAvatar-square css-15w0b3h-MuiAvatar-root"
  >
    <img
      alt="test link name"
      className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
      src="./img/uber-eats-icon.png"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    data-testid="link-name"
  >
    test link name
  </h6>
</a>
`;
